### ISSUE TITLE
Always use Policy-based block BaseFee

### ISSUE THREAD

**Issue by AnnaShaleva** [bane-labs/go-ethereum#329]
# Rationale

f06dad44ee26d71688d2ea356919c36893cfbf09 (#328) removes NeoXBurn fork and makes block's BaseFee to be Policy-based for dBFT-enabled nodes. However, nodes with another consensus engine still use an old-fashioned London way of BaseFee calculation.

# Implementation

Always use Policy-based fee calculation behaviour irrespectively of consensus engine settings. It allows to get rid of some useless code that is never invoked in NeoX. Pitfalls: a lot of tests should be migrated onto new BaseFee calculation scheme or disabled.

**Comment by roman-khimov** [comment]
It's OK for us, NeoX does not have any other consensus mechanism.

**Comment by AnnaShaleva** [comment]
Yes, I know it’s OK, but it requires unit-tests rework, that’s why I moved it to a separate issue.

**Comment by txhsl** [comment]
I tried to run a full test of Geth, and got the following failures.

```
--- FAIL: TestGenerateEncryptedTx (7.74s)
    tpke_test.go:151: 
        	Error Trace:	/home/linux/Desktop/Bane/go-ethereum/antimev/tpke_test.go:151
        	Error:      	Received unexpected error:
        	            	no global public key
        	Test:       	TestGenerateEncryptedTx
FAIL

......

--- FAIL: TestUnderpricingDynamicFee (0.10s)
    legacypool_test.go:1977: adding underpriced pending transaction error mismatch: have <nil>, want transaction underpriced
--- FAIL: TestUnderpricing (0.08s)
    legacypool_test.go:1796: adding underpriced pending transaction error mismatch: have <nil>, want transaction underpriced
--- FAIL: TestDualHeapEviction (0.15s)
    legacypool_test.go:2070: transaction count mismatch: have 30, want 10
FAIL
```

Yes, we need to fix them. But, hmm, if the contract storage of the test chain is empty, then our fee policies should be all `0` so that not affecting the test results, so we don't need to rework them.

OK, so these UT failures are #316 and #372 related, since #316 increased the number of slots.
https://github.com/bane-labs/go-ethereum/blob/28a4058845e358d5f63a2fcd720ba4500b00fb6e/core/txpool/legacypool/legacypool.go#L810-L830
